File: benchmark-stacks-swarm/stacks/apps/java-webflux/src/main/java/com/benchmark/monolith/controller/BonusController.java
package com.benchmark.monolith.controller;

import com.benchmark.monolith.domain.Bonus;
import com.benchmark.monolith.dto.BonusDTO;
import com.benchmark.monolith.service.BonusService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("/bonus")
@RequiredArgsConstructor
public class BonusController {
    private final BonusService service;

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public Mono<Bonus> create(@Valid @RequestBody BonusDTO dto) {
        return service.createBonus(dto);
    }

    @GetMapping("/recents")
    public Flux<Bonus> getRecents() {
        return service.getRecents();
    }

    @GetMapping("/{id}")
    public Mono<Bonus> findOne(@PathVariable Long id) {
        return service.getBonus(id);
    }
}

File: benchmark-stacks-swarm/stacks/apps/node-nestjs/src/bonus/controllers/bonus.controller.ts
import { Controller, Post, Get, Body, Param, HttpCode, HttpStatus } from '@nestjs/common';
import { BonusService } from '../services/bonus.service';
import { Bonus } from '../entities/bonus.entity';
import { CreateBonusDto } from '../dto/create-bonus.dto';

@Controller('bonus')
export class BonusController {
    constructor(private readonly bonusService: BonusService) { }

    @Post()
    @HttpCode(HttpStatus.CREATED)
    create(@Body() bonusData: CreateBonusDto): Promise<Bonus> {
        return this.bonusService.create(bonusData);
    }

    @Get('recents')
    getRecents(): Promise<Bonus[]> {
        return this.bonusService.getRecents();
    }

    @Get(':id')
    findOne(@Param('id') id: string): Promise<Bonus | null> {
        return this.bonusService.findOne(+id);
    }
}
